spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -167,6 +168,10 @@ protected AbstractMessageListenerContainer(ConsumerFactory<? super K, ? super V>
 		if (this.containerProperties.getConsumerRebalanceListener() == null) {
 			this.containerProperties.setConsumerRebalanceListener(createSimpleLoggingConsumerRebalanceListener());
 		}
+		final OffsetCommitCallback commitCallback = containerProperties.getCommitCallback();
+		if (commitCallback != null) {
+			this.containerProperties.setCommitCallback(commitCallback, containerProperties.getOffsetAndMetadataProvider());
```

Comment:
It's a bit on an anti-pattern to name the method `set...` with 2 parameters; in any case, it's not clear why you are re-setting the callback to the same value.

Reply:
Agree, I created a second setter, following the same approach than the callback.